perf/lua/column_insert.lua

```diff
@@ -0,0 +1,147 @@
+--
+-- The test measures run time of batch insertion into the space columns.
+--
+-- Output format (console):
+-- <test-case> <rows-per-second>
+--
+-- NOTE: The test requires a C module. Set the BUILDDIR environment variable to
+-- the tarantool build directory if using out-of-source build.
+--
+
+local clock = require('clock')
+local fiber = require('fiber')
+local fio = require('fio')
+local log = require('log')
+local tarantool = require('tarantool')
+local benchmark = require('benchmark')
+
+local USAGE = [[
+   engine <string, 'memtx'>          - space engine to use for the test
+   wal_mode <string, 'none'>         - write-ahead log mode to use for the test
+   column_count_total <number, 1000> - number of columns in the test space
+   column_count_batch <number, 10>   - number of columns in the record batch
+   row_count_total <number, 1000000> - number of inserted rows
+   row_count_batch <number, 1000>    - number of rows per record batch
+   use_arrow_api <boolean, false>    - use the Arrow API for batch insertion
+
+]]
+
+local params = benchmark.argparse(arg, {
+    {'engine', 'string'},
+    {'wal_mode', 'string'},
+    {'column_count_total', 'number'},
+    {'column_count_batch', 'number'},
+    {'row_count_total', 'number'},
+    {'row_count_batch', 'number'},
+    {'use_arrow_api', 'boolean'},
+}, USAGE)
+
+local DEFAULT_ENGINE = 'memtx'
+local DEFAULT_WAL_MODE = 'none'
```

Comment:
Seems reasonable to make it "write" by default (the same way it's "memtx" instead of "memcs" by default).

Just toughts, feel free to ignore this.

Reply:
Done, with the following changed:
```diff
 box.cfg({
     log = 'tarantool.log',
+    work_dir = test_dir,
     wal_mode = params.wal_mode,
     memtx_memory = 4 * 1024 * 1024 * 1024,
     checkpoint_count = 1,
@@ -144,4 +166,5 @@ end
 bench:dump_results()

 test_module.fini()
+rmtree(test_dir)
 os.exit(0)
```

```diff
@@ -28,11 +28,14 @@ insert_serial_lua_func(struct lua_State *L)
 {
        uint32_t space_id = luaL_checkinteger(L, 1);
        luaL_checktype(L, 2, LUA_TTABLE);
+       lua_getfield(L, 2, "row_count_batch");
+       int batch_row_count = luaL_checkinteger(L, -1);
        lua_getfield(L, 2, "column_count_total");
        int total_column_count = luaL_checkinteger(L, -1);
-       lua_pop(L, 1);
+       lua_pop(L, 2);
        static char tuple_data[1000 * 1000];

+       VERIFY(box_txn_begin() == 0);
        for (int64_t i = 0; i < dataset.row_count; i++) {
                char *data_end = tuple_data;
                data_end = mp_encode_array(data_end, total_column_count);
@@ -48,7 +51,12 @@ insert_serial_lua_func(struct lua_State *L)
                        abort();
                if (box_insert(space_id, tuple_data, data_end, NULL) != 0)
                        return luaT_error(L);
+               if (i % batch_row_count == 0) {
+                       VERIFY(box_txn_commit() == 0);
+                       VERIFY(box_txn_begin() == 0);
+               }
        }
+       VERIFY(box_txn_commit() == 0);
        return 0;
 }
```